addons/barcodes/models/barcode_rule.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+import re
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class BarcodeRule(models.Model):
+    _name = 'barcode.rule'
+    _order = 'sequence asc'
+
+
+    def _get_type_selection(self):
+        return [('alias','Alias'),('product','Unit Product')]
+
+    #indirection for inheritance
+    _type_selection = lambda self: self._get_type_selection()
+
+    name = fields.Char(string='Rule Name', required=True, help='An internal identification for this barcode nomenclature rule')
+    barcode_nomenclature_id = fields.Many2one('barcode.nomenclature', string='Barcode Nomenclature')
+    sequence = fields.Integer(string='Sequence', help='Used to order rules such that rules with a smaller sequence match first')
+    encoding = fields.Selection([
+        ('any', 'Any'),
+        ('ean13', 'EAN-13'),
+        ('ean8', 'EAN-8'),
+        ('upca', 'UPC-A'),
+    ], string='Encoding', required=True, default='any', help='This rule will apply only if the barcode is encoded with the specified encoding')
+    type = fields.Selection(_type_selection, string='Type', required=True, default='product')
```

Comment:
Its my suggestion but python have keyword that's use for reservation :laughing:  hope so you understand my comment , so if really need to change please change it. :space_invader: